drop FreeBSD < 4 support

cremno opened the pull request:
The most recent version affected by this is 3.5 and was released in 2000.

https://www.freebsd.org/releases/3.5R/announce.html
https://en.wikipedia.org/wiki/History_of_FreeBSD#Version_history

nobu commented:
Should we drop the support of older FreeBSD first?

cremno commented:
@nobu: What do you mean? Maybe the title is misleading and I should have written `for FreeBSD < 4`?

nobu commented:
Removing these lines causes test failures.
That is we do not support the older versions anymore.

knu commented:
@nobu Can you point us to the exact test failures this would cause?  I think `defined(__FreeBSD__) && __FreeBSD__ < 4` is always false when it's been over a decade since FreeBSD 3 reached its EOL.

cremno commented:
I've updated the title and added a NEWS entry. Hopefully my intent is more clear now. If this causes tests to fail, I'll fix the PR. But I believe @knu is correct.

nobu commented:
No specific tests may exist.

Only I found is the ChangeLog for 1.4.0:

```
Sun Feb 14 12:47:48 1999  EGUCHI Osamu  <eguchi@shizuokanet.ne.jp>

       * numeric.c (Init_Numeric): allow divide by zero on FreeBSD.
```
